bzira.py

```diff
@@ -100,69 +119,109 @@ def bz_to_jira_priority(options, bug):
+# to query only 1 day's worth of recent changes:
+# https://bugs.eclipse.org/bugs/buglist.cgi?chfieldfrom=1d&status_whiteboard=RHT&order=changeddate%20DESC%2C
+# https://bugs.eclipse.org/bugs/buglist.cgi?chfieldfrom=24h&status_whiteboard=RHT&order=changeddate%20DESC%2C
+# but since that doesn't work (chfieldfrom not supported), do a big query and then filter it for recent changes.
+
+# TODO cache results locally so we don't have to keep hitting live server to do iterations
+bzserver = "https://bugs.eclipse.org/"
+bz = bugzilla.Bugzilla(url=bzserver + "bugs/xmlrpc.cgi")
+if (options.verbose):
+    print "[DEBUG] " + "Querying bugzilla: " + bzserver + "bugs/buglist.cgi?status_whiteboard=RHT"
+query = bz.url_to_query(bzserver + "bugs/buglist.cgi?status_whiteboard=RHT")
 issues = bz.query(query)
-
+if (options.verbose):
+    print "[DEBUG] " + "Found " + str(len(issues)) + " bugzillas to process"
+    print "" 
 
 bugs = []
 
+if (options.verbose):
+    print "[DEBUG] " + "Logging in to " + options.jiraserver
+jira = JIRA(options={'server':options.jiraserver}, basic_auth=(options.username, options.password))
 components = jira.project_components('ERT')
+if (options.verbose):
+    print "[DEBUG] " + "Found " + str(len(components)) + " components in JIRA"
+    print "" 
 
 for bug in issues:
-    #print '%s - %s [%s, %s, [%s]] -> %s' % (bug.id, bug.summary, bug.product, bug.component, bug.target_milestone, bug.weburl)
+    # bug.delta_ts = bugzilla last changed date, eg., 20160106T09:50:33
 
-    issue_dict = create_proxy_jira_dict(options, bug)
+    # print '%s - %s [%s, %s, [%s]] {%s} -> %s' % (bug.id, bug.summary, bug.product, bug.component, bug.target_milestone, bug.delta_ts, bug.weburl)
 
-   ## ensure the product name exists as a component
-    if(not next((c for c in components if bug.product == c.name), None)): 
-        comp = jira.create_component(bug.product, "ERT")
-        components = jira.project_components('ERT')
-    
-    proxyissue = lookup_proxy(options, bug)
-    
-    if(proxyissue):
-        print str(bug.id) + " already proxied at " + proxyissue['key']
-        #print str(proxyissue)
-        fields = {}
-        if (not next((c for c in proxyissue['fields']['components'] if bug.product == c['name']), None)):
-            #TODO: this check for existence in list of components
-            # but then overwrites anything else. Problematic or not ?
-            updcomponents = [{"name" : bug.product}]
-            fields["components"] = updcomponents
-
-        if len(fields)>0:
-            print "Updating " + proxyissue['key'] + " with " + str(fields)
-            isbug = jira.issue(proxyissue['key'])
-            isbug.update(fields)
-            
+    changeddate = datetime.strptime(str(bug.delta_ts), '%Y%m%dT%H:%M:%S')
+    difference = now - changeddate
+
+    if (difference.days > int(options.minimum_age_to_process)):
```

Comment:
One caveat here is that this would mean if we add RHT keyword to old bug bug it would not be detected. Not sure we can avoid that besides running a full query from time to time.